### PR TITLE
LPS-158214 Move creation of OpenAPISchemaFilter in order to be merged with the one created for Objects

### PR DESCRIPTION
**Pull request by ccorreagg**
This PR contains the code that creates the `OpenAPISchemaFilter` based on the Extended Properties in a method that is called, not only from the Liferay System APIs, but also from the resource that creates the OpenAPI for Custom Objects.

With that, the extended properties will be added, not only for the elements of the Liferay System APIs, but also for any Custom Object in case there is an ExtensionProvider that is applicable for the Custom Object.

---

All the OpenAPIs for any Liferay System APIs (for example, headless-delivery) are creating, if applies, an `OpenAPISchemaFilter` with the Extended Properties.

**Before the PR:**

- The OpenAPI generated for Custom Objects uses the `OpenAPISchemaFilter` provided with the transformations generated for each Custom Object, but it does not create either use any extra OpenAPISchemaFilter with the Extended Properties.

**After the PR:**

- The OpenAPI generated for Custom Objects uses the `OpenAPISchemaFilter` provided with the transformations generated for each Custom Object and is merged with the `OpenAPISchemaFilter` created with the Extended Properties in order to apply all the transformation to the initial OpenAPI

### CODE REVIEW

File: modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/resource/OpenAPIResourceImpl.java [hunk 1]
```diff
@@ -119,7 +119,37 @@ public void setConfiguration(
 
 		OpenAPI openAPI = openApiContext.read();
 
+		OpenAPISchemaFilter schemaFilter = null;
```
Comment (javierdearcos) [inline modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/resource/OpenAPIResourceImpl.java:122]: I find the code difficult to follow with so many related conditions. Can we reorganize it to improve readability?

I see some options but I don't see a great one:
- Extract a _mergeOpenAPISchemaFilter method that handles the creation of the final OpenAPISchemaFilter to use and get rid of some conditions.
- Initialize the schemaFilter to the openAPISchemaFilter and get rid of the else and maybe some conditions.
- Reorganize the code so we can have some early returns with an error or basic response like when dtoClassNames is empty.
- Use some sort of Null Object pattern, maybe with OpenAPISchemaFilter.

I think a lot of these variables that we check for null are related and we can improve the code by somehow being aware of these relationships.

**Review by javierdearcos — APPROVED** [review]
Just put a comment to think about in case you think it makes sense and it is possible.

File: modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/resource/OpenAPIResourceImpl.java (hunk 1 above)
Comment (ccorreagg) [reply modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/resource/OpenAPIResourceImpl.java]: I have updated the commit with some of the suggestions. Please, have a look at it.

Any suggestion is welcomed again!

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

File: modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/resource/OpenAPIResourceImpl.java (hunk 1 above)
Comment (javierdearcos) [reply modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/resource/OpenAPIResourceImpl.java]: I like it much more like this. I think it has improved readability

Comment (ccorreagg) [reply modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/resource/OpenAPIResourceImpl.java]: Great. You are right, it has improved a lot

**Comment by ccorreagg** [comment]
ci:forward

**Pull request closed without merging**